src/guest_memory.rs

```diff
@@ -117,11 +119,33 @@ impl Display for Error {
 pub struct GuestAddress(pub u64);
 impl_address_ops!(GuestAddress, u64);
 
+/// Wraps a `GuestAddress` that's known to be aligned with respect to `T`.
+pub type AlignedGuestAddress<T> = Aligned<GuestAddress, T>;
+
+impl<T> std::convert::TryFrom<GuestAddress> for AlignedGuestAddress<T> {
```

Comment:
Implementing `impl<Addr, T> TryFrom<Addr> for Aligned<Addr, T>` should make these two more specific implementations unnecessary.